management/virtualmachine/client.go

```diff
@@ -265,7 +266,7 @@ func (vm VirtualMachineClient) DeleteRole(cloudServiceName, deploymentName, role
 		return "", fmt.Errorf(errParamNotSpecified, "roleName")
 	}
 
-	requestURL := fmt.Sprintf(azureRoleURL, cloudServiceName, deploymentName, roleName)
+	requestURL := fmt.Sprintf(azureRoleURLDelete, cloudServiceName, deploymentName, roleName)
```

Comment:
Closing pull request, will raise again after taking care of the comment above. Also will rebase so that auto-merge can work. Thank you @ahmetalpbalkan